sdk/go/common/workspace/plugins.go

```diff
@@ -785,6 +789,19 @@ func (source *fallbackSource) Download(
 
 	// Fallback to get.pulumi.com
 	pulumi := newGetPulumiSource(source.name, source.kind)
+
+	// Check if there's a URL override for the get.pulumi.com URL
+	if overrideURL, ok := pluginDownloadURLOverridesParsed.get(pulumi.URL()); ok {
```

Comment:
We're already checking this override here: https://github.com/pulumi/pulumi/blob/master/sdk/go/common/workspace/plugins.go#L1387, after we create the fallback source.  So I'm not quite sure why we even need this?